compiler/AST/iterator.cpp

```diff
@@ -544,6 +541,7 @@ buildZip3(IteratorInfo* ii, Vec<BaseAST*>& asts, BlockStmt* singleLoop) {
   }
 
   // Check for more (only for non c for loops)
+  INT_ASSERT(singleLoop);
```

Comment:
I think this assert is ok, but if you think it has value it should probably be added to all the buildZip* functions. I don't think you really need to add the assert since the buildZip* functions are surrounded by an `if (singleLoop)` check, but your call.